crates/config/src/cli.rs

```diff
@@ -0,0 +1,315 @@
+#![deny(missing_docs)]
+//! Command-line interface utilities for configuration management.
+//!
+//! This module provides utility functions for handling configuration overrides
+//! via command-line arguments, supporting the `--set key=value` functionality
+//! across all Hypha binaries.
+
+use figment::providers::{Format, Toml};
+use miette::Result;
+use serde::{Deserialize, Serialize};
+
+use crate::LayeredConfigBuilder;
+
+/// Parse a single key-value pair for configuration overrides.
+///
+/// Parses command-line arguments in the format `KEY=value` into separate
+/// key and value strings for use with configuration override systems.
+///
+/// # Arguments
+///
+/// * `s` - A string slice in the format "key=value"
+///
+/// # Returns
+///
+/// Returns `Ok((key, value))` on successful parsing, or an error message
+/// describing what went wrong.
+///
+/// # Examples
+///
+/// ```rust
+/// use hypha_config::cli::parse_key_val;
+///
+/// let (key, value) = parse_key_val("listen_port=8080").unwrap();
+/// assert_eq!(key, "listen_port");
+/// assert_eq!(value, "8080");
+///
+/// let (key, value) = parse_key_val("debug=true").unwrap();
+/// assert_eq!(key, "debug");
+/// assert_eq!(value, "true");
+///
+/// // Handles nested configuration via dot notation
+/// let (key, value) = parse_key_val("listen_addresses.0=/ip4/127.0.0.1/tcp/8080").unwrap();
+/// assert_eq!(key, "listen_addresses.0");
+/// assert_eq!(value, "/ip4/127.0.0.1/tcp/8080");
+///
+/// // Returns error for invalid format
+/// assert!(parse_key_val("invalid_format").is_err());
+/// ```
+pub fn parse_key_val(s: &str) -> std::result::Result<(String, String), String> {
+    let pos = s
+        .find('=')
+        .ok_or_else(|| format!("invalid KEY=value: no `=` found in `{s}`"))?;
+    Ok((s[..pos].to_string(), s[pos + 1..].to_string()))
+}
```

Comment:
Let's return a `miette::Result` here instead. This will ensure a proper `Error` implementation in case of an error instead of having a string. While technically, there's nothing wrong in using a `String` as error type in a `Result`, it's good practice to use a type that implements the `Error` trait.

```suggestion
pub fn parse_key_val(s: &str) -> Result<(String, String), String> {
    let pos = s
        .find('=')
        .ok_or_else(|| miette::miette!("invalid KEY=value: no `=` found in `{s}`"))?;
    Ok((s[..pos].to_string(), s[pos + 1..].to_string()))
}
```

Reply:
changed 👍